eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/Primitives/KnownManagementTypes.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Resources.Models;
+using Microsoft.TypeSpec.Generator.Primitives;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Azure.Generator.Mgmt.Primitives
+{
+    internal class KnownManagementTypes
+    {
+        private static readonly IReadOnlyDictionary<string, CSharpType> _idToTypeMap = new Dictionary<string, CSharpType>()
+        {
+            ["Azure.ResourceManager.CommonTypes.ExtendedLocation"] = typeof(ExtendedLocation),
+            ["Azure.ResourceManager.CommonTypes.ExtendedLocationType"] = typeof(ExtendedLocationType),
+            ["Azure.ResourceManager.CommonTypes.ManagedServiceIdentity"] = typeof(ManagedServiceIdentity),
+            ["Azure.ResourceManager.CommonTypes.ManagedServiceIdentityType"] = typeof(ManagedServiceIdentityType),
+            ["Azure.ResourceManager.CommonTypes.OperationStatusResult"] = typeof(OperationStatusResult),
+            ["Azure.ResourceManager.CommonTypes.ProxyResource"] = typeof(ResourceData),
+            ["Azure.ResourceManager.CommonTypes.Resource"] = typeof(ResourceData),
+            ["Azure.ResourceManager.CommonTypes.SystemData"] = typeof(SystemData),
+            ["Azure.ResourceManager.CommonTypes.TrackedResource"] = typeof(TrackedResourceData),
+            ["Azure.ResourceManager.CommonTypes.UserAssignedIdentity"] = typeof(UserAssignedIdentity),
+        };
+
+        private static readonly HashSet<string> _azureTypeNamespaceWithName = new()
```

Comment:
Why do we need both the hash set and dictionary?

Reply:
you are right, it's not needed.